### PR TITLE
Replace MLMD HTTP API requests with gRPC-Web requests to ml_metadata

### LINKED ISSUE #148: [MLMD Lineage] Add gRPC Web Client for fetching data from MLMD from gRPC Backend
**Describe the solution you'd like**

Use generated classes for service and API requests in Artifact List and Artifact Details tabs.

**Anything else you would like to add:**

See https://github.com/kubeflow/pipelines/pull/2057 for similar implementation.

/kind feature
/area front-end
/area metadata
/assign @kwasi
/cc @avdaredevil

### PR DESCRIPTION
**Pull request by kwasi**
## Changes

* Add `Api.metadataStoreService` client to request data from `/ml_metadata`
* Proxy all calls to `/ml_metadata*` on metadata-ui frontend sever to `metadata-envoy-proxy` service
* Convert existing calls for Artifacts and Executions to `ml_metadata` calls over gRPC-web 
* Add `npm run start:proxy` to proxy to port 3000 of `metadata-ui` pod. 
  * NOTE: Proxying to port 8080 of `metadata-service` won't work for routing gRPC calls. 

## Dev Changes

* Add mock for grpc-/grpc-web 
* Bump `react-scripts` to 3.2.0 to support `EXTEND_ESLINT` for ignoring generated proto classes
* Add `npm run api:metadata` for generating `grpc-web` protos
* Move jest `__mocks__` into `src/` to work around regression in react-scripts 3.2.0 (https://github.com/facebook/create-react-app/issues/7539)
* Rewrite start_proxy and gen_grpc_web_protos as node scripts.

## NOTE

* This replaces PR #160 which I had a nightmare trying to merge/rebase with intermediate changes pushed in #159 
* Screencast: https://screencast.googleplex.com/cast/NTczOTE4MTYwMDkzMTg0MHw2ODhjMDk0Yy02MQ
* Live Image: https://kubeflow-ui-07e.endpoints.kwasinti-kubeflow-dev.cloud.goog/

Fixes #148

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/metadata/165)

### CODE REVIEW

File: frontend/start-proxy.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  proxyProcess.stdout.on('data', logBuffer);
+  proxyProcess.stderr.on('data', logBuffer);
+  proxyProcess.on('close', code => {
+    console.log(`Metadata UI port forwarding process exited with code ${code}.`)
+  });
+};
+
+detectMetadataUiPodName()
+    .then(podName => portForwardToMetadataUiPod(podName))
+    .catch((reason => {
```
Comment (avdaredevil) [inline frontend/start-proxy.js]:
```suggestion
    .catch(reason => {
      console.log(reason);
    });
```

Comment (kwasi) [reply frontend/start-proxy.js]: Done

**Comment by kwasi** [comment]
Re-tested because the changes seemed unrelated to these changes. 3 gRPC python tests failed with gRPC/port forwarding issues.

```
...
File "/usr/local/lib/python3.5/dist-packages/grpc/_channel.py", line 592, in _end_unary_response_blocking
   raise _Rendezvous(state, None, None, deadline)
grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with:
        status = StatusCode.UNAVAILABLE
        details = "Socket closed"
        debug_error_string = "{"created":"@1573693194.903399374","description":"Error received from peer ipv4:127.0.0.1:8081","file":"src/core/lib/surface/call.cc","file_line":1055,"grpc_
message":"Socket closed","grpc_status":14}"
>
``` 

```
Handling connection for 8081
test_log_invalid_artifacts_should_fail (tests.test.TestMetedata) ... E1114 00:59:54.903145     958 portforward.go:331] an error occurred forwarding 8081 -> 8080: error forwarding port 808
0 to pod 8f89ad4af6465e169bc0b53d74a676481907cbd598998f3b2c9b3bbf3cc8d413, uid : exit status 1: 2019/11/14 00:59:54 socat[6218] E connect(5, AF=2 127.0.0.1:8080, 16): Connection refused
ERROR
```

Full log

http://testing-argo.kubeflow.org/workflows/kubeflow-test-infra/kubeflow-metadata-presubmit-unittests-165-106a22c-6624-a5fe?tab=workflow&nodeId=kubeflow-metadata-presubmit-unittests-165-106a22c-6624-a5fe-3491721471&sidePanel=logs%3Akubeflow-metadata-presubmit-unittests-165-106a22c-6624-a5fe-3491721471%3A

**Pull request merged**